cpp/src/structure/detail/structure_utils.cuh

```diff
@@ -83,6 +83,38 @@ struct rebase_offset_t {
   __device__ edge_t operator()(edge_t offset) const { return offset - base_offset; }
 };
 
+template <typename vertex_t, typename edge_t>
+rmm::device_uvector<vertex_t> expand_sparse_offsets(raft::device_span<edge_t const> offsets,
+                                                    vertex_t base_vertex_id,
+                                                    rmm::cuda_stream_view stream_view)
```

Comment:
Can we use `typename vertex_t` here?
```
      auto graph_ids_v = cugraph::detail::expand_sparse_offsets(
        raft::device_span<size_t const>(d_subgraph_edge_offsets.data(),
                                        d_subgraph_edge_offsets.size()));
                                        d_subgraph_edge_offsets.size()),
        vertex_t{0},
        handle_->get_stream());
```
Here, d_subgraph_edge_offsets.size() can theoretically overflow `vertex_t`, right? # subgraphs to extract can be more than the number of vertices in the graph.

And offsets.back() isn't really the number of edges in this context, but the total number of vertices in all the subgraphs, and this can overflow edge_t as well (may not happen in most cases, but we really can't say this will never happen).

Should we better use either idx_t or size_t for vertex_t and offset_t for edge_t?

Reply:
This is actually just a label in the templated function... I used `vertex_t` and `edge_t` out of habit more than anything else.  `idx_t` and `offset_t` would definitely be better labels for this function.

Changed in the next push.  Also updated some variable names.